src/lib/Nbe.ml

```diff
@@ -814,8 +814,11 @@ struct
         S.Lam bdy
       in 
       S.CodeSg (tbase, tfam)
-    | _, D.CodePath (fam, bdry) ->
-      raise Todo
+    | univ, D.CodePath (fam, bdry) -> (* check *)
+      let* tfam = quote_con univ fam in
+      let+ tbdry = quote_con univ bdry
```

Comment:
Very good guess! The type in which you are quoting the boundary is going to be:

```
    (i : I) -> [i=0\/i=1] -> fam i
```

We can work together to see how to translate that type into code.

Reply:
ah, ok so that's instead of the universe in the second recursive call?

Reply:
Yes, that's right

Reply:
this is now completed based on our session this morning, and with term builder instead of the explicit closures, so i'll "resolve this conversation".